languages/python/oso/tests/regression/test_unhandled_partial.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+from typing import List
+
+def test_unhandled_partial_gh1467():
+    """Test that previously failed due to incorrect partial unification.
+    
+    Submitted by BAH.
+    
+    Fixed in https://github.com/osohq/oso/pull/1467."""
+    POLICY = """\
+actor UserAlchemical {}
+
+allow(actor, action, resource) if has_permission(actor, action, resource);
+
+resource Analytic {
+    permissions = ["Read", "Edit", "Delete", "Share"];
+    roles = ["User", "AnalyticContributor", "Owner", "Admin"];
+
+    # Everyone in the org can read
+    "Read" if "User";
+}
+
+has_role(user: UserAlchemical, "User", analytic: Analytic) if
+    a_role in analytic.groups and
+    "Read" = a_role.permission and
+    a_role.group_id in user.group_ids;
+
+resource AnalyticFilterable {
+    relations = { analytic: Analytic };
+    permissions = ["Read"];
+
+    "Read" if "User" on "analytic";
+}
+
+has_relation(subject: Analytic, "analytic", object: AnalyticFilterable) if
+    subject = object;
+    """
+
+    from dataclasses import dataclass
+
+    from oso import Oso, Variable
+    from polar import Pattern, Expression
+
+    @dataclass
+    class Group:
+        permission: str
+        group_id: int
+
+    @dataclass
+    class Analytic:
+        groups: List[Group]
+
+    @dataclass
+    class AnalyticFilterable(Analytic):
+        pass
+
+    @dataclass
+    class UserAlchemical:
+        group_ids: List[int]
+
+    oso = Oso()
+    oso.register_class(AnalyticFilterable)
+    oso.register_class(Analytic)
+    oso.register_class(UserAlchemical)
+
+    oso.load_str(POLICY)
+
+    constraint = Expression(
+        "And",
+        [Expression("Isa", [Variable("resource"), Pattern("AnalyticFilterable", {})])],
+    )
+
+    # Unhandled partial was raised here before fix.
+    results = list(
+        oso.query_rule(
+            "allow",
+            UserAlchemical(group_ids=[0]),
+            "Read",
+            Variable("resource"),
+            accept_expression=True,
+            bindings={"resource": constraint},
+        )
+    )
+    print(results)
```

Comment:
If it's possible to add a partial test to the core that triggers this issue then we should remove this. If not we should at least change the names and remove the extra details like the unused roles etc.

Reply:
+1

Reply:
I worked a bit on this and couldn't get a simpler repro in the core. I don't have more time to spend on it right now but I do agree that would be better.

Changed the names and simplified this instead.